designdocs/user-handles.md

```diff
@@ -0,0 +1,326 @@
+# Frequency User Handles Registry
+
+## Context and Scope
+
+The Frequency blockchain aims to provide a mechanism to register and retrieve user handles on chain to make it easier to use MSA Ids.
+To achieve this, we propose the creation of a user handle ```registry``` system on frequency chain, coupled with ```MessageSourceAccount```, that allows users to choose a handle and select a suffix from a set of options allowed by the chain that together are unique.
+The user handle system will also enforce guidelines to ensure that handles are user-friendly and easy to remember, as well as prevent misuse and abuse.
+
+The goal of this proposal is to enable users to create unique handles on the Frequency blockchain using a registry, making it more accessible for users to engage with the network.
+
+## Assumptions
+
+* ```MessageSourceAccount``` exists for a user to create a handle.
+* It is optional for a user to create a handle.
+* Handles are unique and has a 1:1 mapping with ```MessageSourceAccount```.
+* Governance will define the range of suffixes allowed for a given handle.
+* Governance will define the time period a handle must be retired before it can be reused.
+
+## Requirements
+
+The high level requirements for user handles are:
+
+* Support user ability to choose any valid handle and select a suffix from a set of options allowed by the chain.
+* Ensure handles and suffixes are unique and non-conflicting.
+* Make the system resistant to namespace exhaustion, homoglyphs and race conditions.
+* Make the system easy to use and integrate with existing UI and wallet systems.
+* Reduce the likelihood of abuse and misuse of the system for example secondary market trading of handles.
+
+## Proposal
+
+User handle ```registry``` on frequency chain.
+
+### Handle Guidelines
+
+* User handles must be unique and each msa can only be be mapped to one handle.
+* User handles must be between 2 and 32 characters (32b limit) long.
+* User handles cannot contain the following substrings (or homoglyphs): @, #, :, ., ```
```

Comment:
Are we thinking of following http://unicode.org/reports/tr46/ similar to ENS?

https://docs.ens.domains/contract-api-reference/name-processing